nautobot/ipam/views.py

```diff
@@ -1214,68 +1215,105 @@ class VLANUIViewSet(NautobotUIViewSet):  # 3.0 TODO: remove, unused BulkImportVi
+            ),
+            PrefixObjectsTablePanel(
+                weight=100,
+                section=SectionChoices.FULL_WIDTH,
+                table_class=tables.PrefixTable,
+                select_related_fields=["status", "role", "namespace"],
+                table_filter="vlan_id",
+                exclude_columns=["vlan"],
+                hide_hierarchy_ui=True,
+            ),
+            object_detail.ObjectsTablePanel(
+                weight=200,
+                section=SectionChoices.FULL_WIDTH,
+                table_class=LocationTable,
+                table_filter="vlans",
+                add_button_route=None,
+            ),
+        ),
+        extra_tabs=(
+            object_detail.DistinctViewTab(
+                weight=100,
+                label="Device Interfaces",
+                url_name="ipam:vlan_device_interfaces",
+                tab_id="device_interfaces",
+                related_object_attribute="interfaces",
+                panels=(
+                    object_detail.ObjectsTablePanel(
+                        weight=100,
+                        section=SectionChoices.FULL_WIDTH,
+                        table_title="Device Interfaces",
+                        table_class=tables.VLANDevicesTable,
+                        table_filter=["untagged_vlan", "tagged_vlans"],
+                        select_related_fields=["device"],
+                        related_field_name="vlan_id",
+                    ),
+                ),
+            ),
+            object_detail.DistinctViewTab(
+                weight=100,
+                label="VM Interfaces",
+                url_name="ipam:vlan_vm_interfaces",
+                tab_id="vm_interfaces",
+                related_object_attribute="vminterfaces",
+                panels=(
+                    object_detail.ObjectsTablePanel(
+                        weight=100,
+                        section=SectionChoices.FULL_WIDTH,
+                        table_title="Virtual Machine Interfaces",
+                        table_class=tables.VLANVirtualMachinesTable,
+                        table_filter=["untagged_vlan", "tagged_vlans"],
+                        select_related_fields=["virtual_machine"],
+                        related_field_name="vlan_id",
+                    ),
+                ),
+            ),
+        ),
+    )
 
-        paginate = {
-            "paginator_class": EnhancedPaginator,
-            "per_page": get_paginate_count(request),
-        }
-        RequestConfig(request, paginate).configure(members_table)
+    @action(
+        detail=True,
+        url_path="device-interfaces",
+        url_name="device_interfaces",
+        custom_view_base_action="view",
+        custom_view_additional_permissions=["dcim.view_interface"],
+    )
+    def device_interfaces(self, request, *args, **kwargs):
+        return Response({})
```

Comment:
For core: we should document somewhere (perhaps done already) why this pattern is needed. I don't understand it reading it, and I am sure I would run into the source of the issue if I came into it.